### PR TITLE
bugs fixing with cron and notification implementation

### PR DESCRIPTION
**Pull request by alinagoal**

### CODE REVIEW

File: controllers/v1/api/driverController.js [hunk 1]
```diff
@@ -107,19 +109,22 @@ class driverController extends controller {
 
     async updateTruckStatusAndLocation(req, res, next) {
         try {
-            let { locationStationed, status, currentLocation } = req.body
+            let { locationStationed, status, currentLocation,lastSyncedDate } = req.body
```
Comment (Nileshmaharjan) [inline controllers/v1/api/driverController.js:112]: lastsyncDate = new Date () yesto garnu backend ma matra!  request body ma halirakhnu pardaina

File: controllers/v1/api/driverController.js [hunk 2]
```diff
@@ -107,19 +109,22 @@ class driverController extends controller {
...
+            console.log(' req.body:::::::', req.body)
+            console.log(' req.req.user._id:::::::', req.user._id)
```
Comment (Nileshmaharjan) [inline controllers/v1/api/driverController.js:114]: remove console logs everywhere

File: controllers/v1/api/driverController.js [hunk 3]
```diff
@@ -154,14 +159,15 @@ class driverController extends controller {
                 currentLocation: ((currentLocation && currentLocation.length > 0) ? currentLocation : truckExist.currentLocation),
                 locationStationed: ((locationStationed && locationStationed.coordinates.length > 0) ? {
                     coordinates: coordinates
-                } : truckExist.locationStationed)
+                } : truckExist.locationStationed),
+                lastSyncedDate: lastSyncedDate
```
Comment (Nileshmaharjan) [inline controllers/v1/api/driverController.js:163]: lastSyncedDate: new Date () change yesma

File: controllers/v1/api/driverController.js [hunk 4]
```diff
@@ -183,6 +189,42 @@ class driverController extends controller {
             next(error)
         }
     }
+
+    async saveRegistrationToken(req, res, next) {
+        try {
+
+        const {
+            driverId,
```
Comment (Nileshmaharjan) [inline controllers/v1/api/driverController.js:197]: driverId request garnu pardaina, It should come at req.user._id ma ! check console.log(req.user._id) or console.log(req.driver._id)

File: cron.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const cron = require('node-cron');
+const cronService = require('./modules/fcmService')
+
+const performCron = () => {
+
+    console.log('start cron job::::::::::' );
+    
+    const fcmCron = new cronService()
+    let task = cron.schedule('*/30 7-21 * * *', async () => {
```
Comment (Nileshmaharjan) [inline cron.js:9]: cron check gareko ho?

File: models/truck.js [hunk 6]
```diff
@@ -243,6 +243,10 @@ const truckSchema = new Schema({
         type: Date,
         default: null
     },
+    lastSyncedDate: {
+        type: Number,
```
Comment (Nileshmaharjan) [inline models/truck.js:247]: model type: Date hunchha

File: models/driverNotifications.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+const { ObjectId } = require('../helpers/objectIdHelper');
+const mongooseAggregatePaginate = require('mongoose-aggregate-paginate-v2');
+const mongoosePaginate = require('mongoose-paginate-v2');
+const enumVars = require('../config/enumVariables');
+
+const driverNotificationSchema = new mongoose.Schema({
+    driver: { type: mongoose.Schema.Types.ObjectId, ref: 'driver', required: true },
+    title: { type: String, required: true },
+    body: { type: String, required: true },
```
Comment (Nileshmaharjan) [inline models/driverNotifications.js:11]: yo model use bhako chhaina hai aile?

File: repositories/truckRepository.js [hunk 8]
```diff
@@ -138,7 +138,40 @@ module.exports = {
...
+
+        //         }}
+        // ])
+        // if (typeof options.search === 'string' && options.search.length > 0) {
+        //     // console.log('driversFromSearch',driversFromSearch)
+        //     query.$match.$and.push({
+        //      $or: [{ "vehicle.truckNumber": { $regex: options.search, $options: 'i' } },
+        //         //  { "driver": { $in: driversFromSearch.map(item => item['_id']),  } }],
+        //      ]})
+        // }
```
Comment (Nileshmaharjan) [inline repositories/truckRepository.js:174]: line number 182 ko query lai $project stage pachi halnu

Also change to 

           query.$match.$and.push({
                $or: [{ "vehicle.truckNumber": { $regex: options.search, $options: 'i' } }, { "driver.firstName": { $regex: options.search, $options: 'i' } }],

File: controllers/v1/api/driverController.js (hunk 4 above)
Comment (alinagoal) [reply controllers/v1/api/driverController.js:197]: Audaina dai. undefined vancha.

File: cron.js (hunk 5 above)
Comment (alinagoal) [reply cron.js:9]: yes check gareko ho

File: models/truck.js (hunk 6 above)
Comment (alinagoal) [reply models/truck.js:247]: date kina? mailay epich time sanga check gareko. date garda tdat type error airako thyo so.

File: models/driverNotifications.js (hunk 7 above)
Comment (alinagoal) [reply models/driverNotifications.js:11]: use chai vako chaina. I'll delete it. Thanks

**Review by bits-bytes-2021 — APPROVED** [review]

**Pull request merged**